wandb/sdk/artifacts/artifact.py

```diff
@@ -1590,23 +1622,48 @@ def _download_file(
             _thread_local_api_settings.cookies = cookies
             _thread_local_api_settings.headers = headers
 
-            self.get_path(name).download(root)
+            entry.download(root)
             download_logger.notify_downloaded()
 
-        pool = multiprocessing.dummy.Pool(32)
-        pool.map(
-            partial(
-                _download_file,
-                api_key=_thread_local_api_settings.api_key,
-                cookies=_thread_local_api_settings.cookies,
-                headers=_thread_local_api_settings.headers,
-            ),
-            self.manifest.entries,
+        download_entry = partial(
+            _download_entry,
+            api_key=_thread_local_api_settings.api_key,
+            cookies=_thread_local_api_settings.cookies,
+            headers=_thread_local_api_settings.headers,
         )
+
+        with concurrent.futures.ThreadPoolExecutor(64) as executor:
```

Comment:
This block is sufficiently complex conceptually to merit an explanatory comment. I would even prefer to break it out into a helper function just so this function reads a little more directly, but that's not the style of the rest of the code in this file.

Reply:
Agreed. I actually plan to move this to a separate `ArtifactDownloader` class in a later PR.